bigbone/src/main/kotlin/social/bigbone/api/method/PushNotificationMethods.kt

```diff
@@ -0,0 +1,137 @@
+package social.bigbone.api.method
+
+import social.bigbone.MastodonClient
+import social.bigbone.MastodonRequest
+import social.bigbone.Parameters
+import social.bigbone.api.entity.WebPushSubscription
+import social.bigbone.api.exception.BigBoneRequestException
+
+/**
+ * Allows access to API methods with endpoints having an "api/vX/push" prefix.
+ * @see <a href="https://docs.joinmastodon.org/methods/push/">Mastodon push notification API methods</a>
+ */
+class PushNotificationMethods(private val client: MastodonClient) {
+
+    private val pushEndpoint = "/api/v1/push/subscription"
+
+    /**
+     * Specify whether to receive push notifications from all, followed, follower, or none users.
+     */
+    enum class PushDataPolicy {
+        ALL,
+
+        FOLLOWED,
+
+        FOLLOWER,
+
+        NONE
+    }
+
+    /**
+     * Add a Web Push API subscription to receive notifications.
+     * Each access token can have one push subscription. If you create a new subscription, the old subscription is deleted.
+     * @see <a href="https://docs.joinmastodon.org/methods/push/#v1">Mastodon API documentation: methods/push/#v1</a>
+     */
```

Comment:
I think it could make sense to add info about the meaning of the parameters using `@param` kDoc entries here and for the other methods.

It’s not clear, unless looking at the linked documentation, that for example `userPublicKey` should be “User agent public key. Base64 encoded string of a public key from a ECDH keypair using the prime256v1 curve.”

Let’s add the info available in the documentation to the documentation here, but without the type as that is already visible through the Kotlin types.